client/src/Editor.jsx

```diff
@@ -0,0 +1,226 @@
+import React, { useEffect, useState, useRef} from 'react';
+import { Uri, editor, KeyMod, KeyCode, languages } from 'monaco-editor';
+import { initializeMode } from 'monaco-graphql/esm/initializeMode';
+import { createGraphiQLFetcher } from '@graphiql/toolkit';
+import * as JSONC from 'jsonc-parser';
+import { debounce } from './utils/debounce';
+
+// description that is displayed in request pane above actual code
+// indicate to the user what commands are available and defined
+// see below: 'queryAction'
+const defaultOperations =
+  localStorage.getItem('operations') ??
+  `
+# GQL Request Pane
+# cmd/ctrl + return/enter will execute the operation
+# Also available via context menu & f1 command palette
+
+query {
+
+}
+`;
+
+// same as above but for the variables pane
+const defaultVariables =
+  localStorage.getItem('variables') ??
+  `
+/* Variables Pane
+cmd/ctrl + return/enter will execute the operation
+Format your variables as valid JSON */
+
+{
+
+}
+`;
+
+// docs here are sparse
+// i believe that the URI here is *internally* looking into monaco config files
+// to find the proper model which is specified as an argument.
+// if a URI such as 'operation/graphql' is passed in and is resolved to a real model,
+// then that model will be returned. (model being the document utilized in the editor)
+// otherwise, the model is created
+const getOrCreateModel = (uri, value,) => {
+  return (
+    editor.getModel(Uri.file(uri)) ??
+    editor.createModel(value, uri.split('.').pop(), Uri.file(uri))
+  );
+};
+
+
+// set these early on so that initial variables with comments don't flash an error
+// NOTE: This may set up options for the variables being cross checked across the schema
+// may need more here
+languages.json.jsonDefaults.setDiagnosticsOptions({
+  allowComments: true,
+  trailingCommas: 'ignore',
+});
+
+
+// add editor to DOM
+// ref is the element the editor will hook onto
+const createEditor = (
+  ref,
+  options,
+) => editor.create(ref.current, options);
+
+
+export default function Editor({schema, endpoint}) {
```

Comment:
The only thing I understand well enough to ask the question is, what are the team's feelings of default exports vs named exports?

Reply:
I'm neutral, I left the boilerplate the way it was for this :)